upstash_redis/commands.py

```diff
@@ -427,6 +427,263 @@ def hexpire(
 
         return self.execute(command)
 
+    def hpexpire(
+        self,
+        key: str,
+        fields: Union[str, List[str]],
+        milliseconds: Union[int, datetime.timedelta],
+        nx: bool = False,
+        xx: bool = False,
+        gt: bool = False,
+        lt: bool = False,
+    ) -> ResponseT:
+        """
+        Sets a timeout on a hash field in milliseconds.
+        After the timeout has expired, the hash field will automatically be deleted.
+
+        :param key: The key of the hash.
+        :param fields: The field(s) within the hash to set the expiry for.
+        :param milliseconds: The timeout in milliseconds as an int or a datetime.timedelta object.
+        :param nx: Set expiry only when the field has no expiry.
+        :param xx: Set expiry only when the field has an existing expiry.
+        :param gt: Set expiry only when the new expiry is greater than the current one.
+        :param lt: Set expiry only when the new expiry is less than the current one.
+        """
+        if isinstance(milliseconds, datetime.timedelta):
+            milliseconds = int(milliseconds.total_seconds() * 1000)
+
+        command: List = ["HPEXPIRE", key, milliseconds]
+
+        if nx:
+            command.append("NX")
+        if xx:
+            command.append("XX")
+        if gt:
+            command.append("GT")
+        if lt:
+            command.append("LT")
+
+        if isinstance(fields, str):
+            fields = [fields]
+
+        command.extend(["FIELDS", len(fields), *fields])
+
+        return self.execute(command)
+
+    def hexpireat(
+        self,
+        key: str,
+        fields: Union[str, List[str]],
+        unix_time_seconds: Union[int, datetime.datetime],
+        nx: bool = False,
+        xx: bool = False,
+        gt: bool = False,
+        lt: bool = False,
+    ) -> ResponseT:
+        """
+        Set an expiration time for specific fields in a hash at a given Unix timestamp.
+
+        Args:
+            key (str): The key of the hash.
```

Comment:
the documentation structure is not consistent with the rest of the client, see `hpexpire` for the preferred way